tests/failpoints/cases/test_split_region.rs

```diff
@@ -63,6 +67,68 @@ fn test_follower_slow_split() {
     assert!(rx.recv_timeout(Duration::from_millis(100)).is_ok());
 }
 
+fn gen_split_region() -> (Region, Region, Region) {
+    let mut cluster = new_server_cluster(0, 2);
+    let region_max_size = 50000;
+    let region_split_size = 30000;
+    cluster.cfg.raft_store.split_region_check_tick_interval = ReadableDuration::millis(100);
+    cluster.cfg.coprocessor.region_max_size = ReadableSize(region_max_size);
+    cluster.cfg.coprocessor.region_split_size = ReadableSize(region_split_size);
+
+    let mut range = 1..;
+    cluster.run();
+    let pd_client = Arc::clone(&cluster.pd_client);
+    let region = pd_client.get_region(b"").unwrap();
+    let last_key = put_till_size(&mut cluster, region_split_size, &mut range);
+    let target = pd_client.get_region(&last_key).unwrap();
+
+    assert_eq!(region, target);
+
+    let max_key = put_cf_till_size(&mut cluster, CF_WRITE, region_max_size, &mut range);
+
+    let left = pd_client.get_region(b"").unwrap();
+    let right = pd_client.get_region(&max_key).unwrap();
+    if left == right {
+        cluster.wait_region_split_max_cnt(&region, 20, 250, false);
+    }
+
+    let left = pd_client.get_region(b"").unwrap();
+    let right = pd_client.get_region(&max_key).unwrap();
+
+    (region, left, right)
+}
+
+#[test]
+fn test_pause_split_when_snap_gen_will_split() {
+    let is_generating_snapshot = "is_generating_snapshot";
+    fail::cfg(is_generating_snapshot, "return()").unwrap();
+
+    let (region, left, right) = gen_split_region();
+
+    assert_ne!(left, right);
+    assert_eq!(region.get_start_key(), left.get_start_key());
+    assert_eq!(right.get_start_key(), left.get_end_key());
+    assert_eq!(region.get_end_key(), right.get_end_key());
+
+    fail::remove(is_generating_snapshot);
+}
+
+#[test]
+fn test_pause_split_when_snap_gen_never_split() {
```

Comment:
The case can run too long, better use smaller timeout.

Reply:
I'll change to try 50 times, 20ms timeout, which is 1 second.

Reply:
1 second is still too long. You can adapt `gen_split_region` to accept custom timeout. 200ms should be enough for one split.

Reply:
got it.